src/main/java/com/example/animalshelter/model/Animal.java

```diff
@@ -1,25 +1,57 @@
 package com.example.animalshelter.model;
 
-import lombok.Builder;
 import lombok.Data;
 
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Positive;
+import javax.persistence.*;
 
 /**
  * Keeps basic information about an animal
  */
 @Data
```

Comment:
For now, it should not be a problem as there is no `@ManyToMany` etc. annotation but also you should be careful with Lombok and JPA. [Here](https://www.jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/) more about it.

Reply:
Thanks for pointing this out and for the link!

Reply:
As one of the authors of this article, I can say that even without @ManyToMany @Data makes this code wrong. Equals and Hashcode should include only id, and in this case, it includes all fields... So, you will not be able to use it in HashSet or HashMap (as a key).

Reply:
Thank you @aleksey-stukalov for your input! I have removed @Data annotation.